### PR TITLE
Fixed aws_dlm_lifecycle_policy table fails with error on querying Closes #1830

### LINKED ISSUE #1830: Table `aws_dlm_lifecycle_policy` fails with error on querying
**Describe the bug**
```
select * from aws_dlm_lifecycle_policy

Error: operation error DLM: GetLifecyclePolicy, https response error StatusCode: 200, RequestID: 69a46b98-9de1-443a-8b12-8636aab7029e, deserialization failed, failed to decode response body with invalid JSON, expected Timestamp to be a JSON Number, got string instead (SQLSTATE HV000)

+-----------+-----+-------------+--------------+---------------+--------------------+-------------+-------+----------------+----------------+------+-------+------+-----------+--------+------------+------+
| policy_id | arn | description | date_created | date_modified | execution_role_arn | policy_type | state | status_message | policy_details | tags | title | akas | partition | region | account_id | _ctx |
+-----------+-----+-------------+--------------+---------------+--------------------+-------------+-------+----------------+----------------+------+-------+------+-----------+--------+------------+------+
+-----------+-----+-------------+--------------+---------------+--------------------+-------------+-------+----------------+----------------+------+-------+------+-----------+--------+------------+------+

```

**Steampipe version (`steampipe -v`)**
Example: v0.20.8

**Plugin version (`steampipe plugin list`)**
Example: v0.109.1

**To reproduce**
Steps to reproduce the behavior (please include relevant code and/or commands).

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_dlm_lifecycle_policy []

PRETEST: tests/aws_dlm_lifecycle_policy

TEST: tests/aws_dlm_lifecycle_policy
Running terraform
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_caller_identity.current: Read complete after 2s [id=222222222222]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_dlm_lifecycle_policy.named_test_resource will be created
  + resource "aws_dlm_lifecycle_policy" "named_test_resource" {
      + arn                = (known after apply)
      + description        = "example DLM lifecycle policy"
      + execution_role_arn = (known after apply)
      + id                 = (known after apply)
      + state              = "DISABLED"
      + tags_all           = (known after apply)

      + policy_details {
          + policy_type        = "EBS_SNAPSHOT_MANAGEMENT"
          + resource_locations = (known after apply)
          + resource_types     = [
              + "VOLUME",
            ]
          + target_tags        = {
              + "Snapshot" = "true"
            }

          + schedule {
              + copy_tags   = false
              + name        = "2 weeks of daily snapshots"
              + tags_to_add = {
                  + "SnapshotCreator" = "DLM"
                }

              + create_rule {
                  + interval      = 24
                  + interval_unit = "HOURS"
                  + location      = (known after apply)
                  + times         = [
                      + "23:45",
                    ]
                }

              + retain_rule {
                  + count = 14
                }
            }
        }
    }

  # aws_iam_role.named_test_resource will be created
  + resource "aws_iam_role" "named_test_resource" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = "dlm.amazonaws.com"
                        }
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + managed_policy_arns   = (known after apply)
      + max_session_duration  = 3600
      + name                  = "turbottest78823"
      + name_prefix           = (known after apply)
      + path                  = "/"
      + tags_all              = (known after apply)
      + unique_id             = (known after apply)
    }

  # aws_iam_role_policy.named_test_resource will be created
  + resource "aws_iam_role_policy" "named_test_resource" {
      + id     = (known after apply)
      + name   = "turbottest78823"
      + policy = jsonencode(
            {
              + Statement = [
                  + {
                      + Action   = [
                          + "ec2:CreateSnapshot",
                          + "ec2:CreateSnapshots",
                          + "ec2:DeleteSnapshot",
                          + "ec2:DescribeInstances",
                          + "ec2:DescribeVolumes",
                          + "ec2:DescribeSnapshots",
                        ]
                      + Effect   = "Allow"
                      + Resource = "*"
                    },
                  + {
                      + Action   = [
                          + "ec2:CreateTags",
                        ]
                      + Effect   = "Allow"
                      + Resource = "arn:aws:ec2:*::snapshot/*"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + role   = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "222222222222"
  + aws_partition = "aws"
  + aws_region    = "us-east-1"
  + resource_aka  = (known after apply)
  + resource_id   = (known after apply)
  + resource_name = "turbottest78823"
aws_iam_role.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creation complete after 2s [id=turbottest78823]
aws_iam_role_policy.named_test_resource: Creating...
aws_dlm_lifecycle_policy.named_test_resource: Creating...
aws_iam_role_policy.named_test_resource: Creation complete after 1s [id=turbottest78823:turbottest78823]
aws_dlm_lifecycle_policy.named_test_resource: Creation complete after 3s [id=policy-00ba99feb94af8713]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 43, in data "null_data_source" "resource":
  43: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "222222222222"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:dlm:us-east-1:222222222222:policy/policy-00ba99feb94af8713"
resource_id = "policy-00ba99feb94af8713"
resource_name = "turbottest78823"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:dlm:us-east-1:222222222222:policy/policy-00ba99feb94af8713",
    "policy_id": "policy-00ba99feb94af8713"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:aws:dlm:us-east-1:222222222222:policy/policy-00ba99feb94af8713",
    "policy_id": "policy-00ba99feb94af8713"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:dlm:us-east-1:222222222222:policy/policy-00ba99feb94af8713",
    "policy_id": "policy-00ba99feb94af8713"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "222222222222",
    "akas": [
      "arn:aws:dlm:us-east-1:222222222222:policy/policy-00ba99feb94af8713"
    ],
    "partition": "aws",
    "region": "us-east-1",
    "title": "policy-00ba99feb94af8713"
  }
]
✔ PASSED

POSTTEST: tests/aws_dlm_lifecycle_policy

TEARDOWN: tests/aws_dlm_lifecycle_policy

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_dlm_lifecycle_policy
+--------------------------+--------------------------------------------------------------------+------------------------------+---------------------------+---------------------------+------------------------------------------------+-------------------------+---------->
| policy_id                | arn                                                                | description                  | date_created              | date_modified             | execution_role_arn                             | policy_type             | state    >
+--------------------------+--------------------------------------------------------------------+------------------------------+---------------------------+---------------------------+------------------------------------------------+-------------------------+---------->
| policy-05abda7dbeff8eaf3 | arn:aws:dlm:us-east-1:222222222222:policy/policy-05abda7dbeff8eaf3 | example DLM lifecycle policy | 2022-08-30T21:56:05+05:30 | 2022-08-30T21:56:05+05:30 | arn:aws:iam::222222222222:role/turbottest7624  | EBS_SNAPSHOT_MANAGEMENT | DISABLED >
| policy-0cc0fb53caf6dc133 | arn:aws:dlm:us-east-1:222222222222:policy/policy-0cc0fb53caf6dc133 | example DLM lifecycle policy | 2022-09-02T10:16:26+05:30 | 2022-09-02T10:16:26+05:30 | arn:aws:iam::222222222222:role/turbottest59347 | EBS_SNAPSHOT_MANAGEMENT | DISABLED >
| policy-03d92da37bc1639e5 | arn:aws:dlm:us-east-1:222222222222:policy/policy-03d92da37bc1639e5 | example DLM lifecycle policy | 2022-08-30T20:14:29+05:30 | 2022-08-30T20:14:29+05:30 | arn:aws:iam::222222222222:role/turbottest87200 | EBS_SNAPSHOT_MANAGEMENT | DISABLED >
+--------------------------+--------------------------------------------------------------------+------------------------------+---------------------------+---------------------------+------------------------------------------------+-------------------------+---------->
```
</details>

### CODE REVIEW

File: aws/table_aws_dlm_lifecycle_policy.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func tableAwsDLMLifecyclePolicy(_ context.Context) *plugin.Table {
 			KeyColumns: plugin.SingleColumn("policy_id"),
 			Hydrate:    getDLMLifecyclePolicy,
 			IgnoreConfig: &plugin.IgnoreConfig{
-				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFound"}),
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFound", "ResourceNotFoundException"}),
```
Comment (misraved) [inline aws/table_aws_dlm_lifecycle_policy.go:27]: @ParthaI do we need to add both the error codes in the ignore error func?

File: go.mod [hunk 2]
```diff
@@ -37,7 +37,7 @@ require (
 	github.com/aws/aws-sdk-go-v2/service/databasemigrationservice v1.23.1
 	github.com/aws/aws-sdk-go-v2/service/dax v1.12.0
 	github.com/aws/aws-sdk-go-v2/service/directoryservice v1.16.0
-	github.com/aws/aws-sdk-go-v2/service/dlm v1.14.2
+	github.com/aws/aws-sdk-go-v2/service/dlm v1.15.6
```
Comment (misraved) [inline go.mod:40]: Were there any breaking changes mentioned in the CHANGELOG for this SDK version update?

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

File: aws/table_aws_dlm_lifecycle_policy.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_dlm_lifecycle_policy.go:27]: @misraved, The API version which was used earlier (`v1.14.2`) that returns `ResourceNotFound`, but as per the current version (`v1.15.6`) the API is returning the `ResourceNotFoundException` if resource is not available.

 Removed the `ResourceNotFound` from ignore config.

**Review by misraved — APPROVED** [review]

**Pull request merged**